### PR TITLE
Kotlin generator migration/cleanup

### PR DESCRIPTION
**Pull request by tcslater**
Cleaning up kotlin related files for migration to https://github.com/swagger-api/swagger-codegen-generators

Relies on related PR for swagger-codegen-generators https://github.com/swagger-api/swagger-codegen-generators/pull/26

### CODE REVIEW

**Comment by jmini** [comment]
The [cleanup tasks](
https://github.com/swagger-api/swagger-codegen/wiki/Swagger-Codegen-migration-(swagger-codegen-generators-repository)#cleanup-work-in-the-swagger-codegen-repository) are now different (this explains the conflicts):

* `io.swagger.codegen.CodegenConfig` is already removed
* In `/bin/kotlin-client-petstore.sh` you need to uncomment the `java` command => `java` instead of `# java`
* In `/bin/windows/kotlin-client-petstore.bat` you need to uncomment the `java` command => `java` instead of `REM java`

**Comment by tcslater** [comment]
Additional cleanup tasks now complete

File: bin/kotlin-client-petstore.sh [hunk 1]
```diff
@@ -26,6 +26,6 @@ fi
 
 # if you've executed sbt assembly previously it will use that instead.
 export JAVA_OPTS="${JAVA_OPTS} -XX:MaxPermSize=256M -Xmx1024M -DloggerPath=conf/log4j.properties"
-ags="generate -t modules/swagger-codegen/src/main/resources/kotlin-client -i modules/swagger-codegen/src/test/resources/2_0/petstore.yaml -l kotlin --artifact-id kotlin-petstore-client -o samples/client/petstore/kotlin $@"
+ags="generate -i modules/swagger-codegen/src/test/resources/2_0/petstore.yaml -l kotlin-client --artifact-id kotlin-petstore-client -o samples/client/petstore/kotlin $@"
```
Comment (jimschubert) [inline bin/kotlin-client-petstore.sh]: I think it's weird that "language" is "kotlin-client".

**Comment by tcslater** [comment]
This is should be good to merge now that swagger-api/swagger-codegen-generators#26 has gone in

**Comment by HugoMario** [comment]
thanks @tcslater !!

**Pull request merged**